hbase-server/src/test/java/org/apache/hadoop/hbase/master/cleaner/TestSeparateOldWALCleaner.java

```diff
@@ -0,0 +1,186 @@
+  @Rule
+  public final TestName name = new TestName();
+
+  public TestSeparateOldWALCleaner()  {
+    this.server = null;
+    this.tableName = null;
+
+    String className = this.getClass().getName();
+    StringBuilder v = new StringBuilder(className);
+    while (v.length() < 1000) {
+      v.append(className);
+    }
+    this.value = Bytes.toBytes(v.toString());
+  }
+
+  // Need to override this setup so we can edit the config before it gets sent
+  // to the HDFS & HBase cluster startup.
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+
+    Configuration conf = TEST_UTIL.getConfiguration();
+    conf.setBoolean("hbase.separate.oldlogdir.by.regionserver",true);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    // Use 2 DataNodes and default values for other StartMiniCluster options.
+    TEST_UTIL.startMiniCluster(StartMiniClusterOption.builder().numDataNodes(2).build());
+
+    cluster = TEST_UTIL.getHBaseCluster();
+    dfsCluster = TEST_UTIL.getDFSCluster();
+    fs = TEST_UTIL.getTestFileSystem();
+    admin = TEST_UTIL.getAdmin();
+
+    // disable region rebalancing (interferes with log watching)
+    cluster.getMaster().balanceSwitch(false);
+  }
+
+  @After
+  public void tearDown() throws Exception  {
+    TEST_UTIL.shutdownMiniCluster();
+  }
+
+  protected void startAndWriteData() throws IOException, InterruptedException {
+    TEST_UTIL.getConnection().getTable(TableName.META_TABLE_NAME);
+    this.server = cluster.getRegionServerThreads().get(0).getRegionServer();
+
+    Table table = createTestTable();
+    server = TEST_UTIL.getRSForFirstRegionInTable(table.getName());
+    for (int i = 1; i <= 32; i++) {
+      doPut(table, i);
+      if (i % 4 == 0) {
+        try {
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
+          // continue
+        }
+      }
+    }
+  }
+
+  /**
+   * Tests that oldWALs/rs are deleted
+   */
+  @Test
+  public void testCleanedRSLogRolling() throws Exception {
+    this.tableName = getName();
+    startAndWriteData();
+    RegionInfo region = server.getRegions(TableName.valueOf(tableName)).get(0).getRegionInfo();
+    final WAL log = server.getWAL(region);
+    LOG.info("there are " + AbstractFSWALProvider.getNumRolledLogFiles(log) + " log files");
+
+    // flush all regions
+    for (HRegion r : server.getOnlineRegionsLocalContext()) {
+      r.flush(true);
+    }
+
+    //start cleaner chore
+    Configuration conf = TEST_UTIL.getConfiguration();
+    Path rootdir = CommonFSUtils.getWALRootDir(conf);
+    Path OLD_WALS_DIR =
+        new Path(rootdir, HConstants.HREGION_OLDLOGDIR_NAME);
+    DirScanPool POOL = new DirScanPool(TEST_UTIL.getConfiguration());
+    LogCleaner cleaner = new LogCleaner(100, server, conf, fs, OLD_WALS_DIR, POOL);
+    Thread thread = new Thread(() -> cleaner.chore());
+    thread.setDaemon(true);
+    thread.start();
+
+    // Now roll the log
+    log.rollWriter();
```

Comment:
We must assert what we are testing in this method. If we want to validate that wal file got deleted from the given RS oldWAL dir, then we should verify that the given folder is now empty.